apolo_kube_client/_core.py

```diff
@@ -0,0 +1,220 @@
+
+    async def post(self, *args: Any, **kwargs: Any) -> dict[str, Any]:
+        resp = await self.request("POST", *args, **kwargs)
+        return cast(dict[str, Any], await resp.json())
+
+    async def patch(self, *args: Any, **kwargs: Any) -> dict[str, Any]:
+        resp = await self.request("PATCH", *args, **kwargs)
+        return cast(dict[str, Any], await resp.json())
+
+    async def put(self, *args: Any, **kwargs: Any) -> dict[str, Any]:
+        resp = await self.request("PUT", *args, **kwargs)
+        return cast(dict[str, Any], await resp.json())
+
+    async def delete(self, *args: Any, **kwargs: Any) -> dict[str, Any]:
+        resp = await self.request("DELETE", *args, **kwargs)
+        return cast(dict[str, Any], await resp.json())
+
+    @property
+    def _auth_headers(self) -> dict[str, Any]:
+        if self._auth_type != KubeClientAuthType.TOKEN or not self._token:
+            return {}
+        return {"Authorization": f"Bearer {self._token}"}
+
+    @staticmethod
+    async def _raise_for_status(response: aiohttp.ClientResponse) -> None:
+        if response.status >= 400:
+            payload = await response.text()
+            match response.status:
+                case 400:
+                    raise ResourceBadRequest(payload)
+                case 401:
+                    raise KubeClientUnauthorized(payload)
+                case 403:
+                    raise KubeClientException(payload)
+                case 404:
+                    raise ResourceNotFound(payload)
+                case 409:
+                    raise ResourceExists(payload)
+                case 410:
+                    raise ResourceGone(payload)
+                case 422:
+                    raise ResourceInvalid(payload)
+                case _:
+                    raise KubeClientException(payload)
+
+    @property
+    def _is_ssl(self) -> bool:
+        return self.base_url.scheme == "https"
+
+    def _create_ssl_context(self) -> ssl.SSLContext | bool:
+        if not self._is_ssl:
+            return False
+        ssl_context = ssl.create_default_context(
+            cafile=self._cert_authority_path, cadata=self._cert_authority_data_pem
+        )
+        if self._auth_type == KubeClientAuthType.CERTIFICATE:
+            ssl_context.load_cert_chain(
+                self._auth_cert_path,  # type: ignore
+                self._auth_cert_key_path,
+            )
+        return ssl_context
+
+    async def _start_token_updater(self) -> None:
+        """
+        A task which periodically reads from the `token_path` and refreshes the token
+        """
+        if not self._token_path:
+            logger.info("%s: token path does not exist. updater won't be started", self)
+            return
+
+        logger.info("%s: starting token updater task", self)
+
+        while True:
+            try:
+                self._refresh_token_from_file()
+            except asyncio.CancelledError:
+                raise
```

Comment:
```suggestion
            except asyncio.CancelledError:
                raise
```
Starting from python 3.9, `CancelledError` is derived from `BaseException`.
Thus, `except Exception` doesn't catch it, a special case for the task cancellation could be omitted in almost all cases.

Reply:
removed this except